grammars/php.cson

```diff
@@ -3724,3 +3734,32 @@
         ]
       }
     ]
+  
+  'ternary_shorthand':
+    'match': '\\s*(\\?:)\\s*'
```

Comment:
Don't capture the spaces, they're not adding anything of value.  Then this can be shortened to `'match': '\\?:'`, and you can scope the 0th capture.

Reply:
Wasn't sure if textmate ate spaces or you had to be explicit with each token. 

To clarify: Textmate _does_ eat spaces when not defined explicitly?

Reply:
I'm not sure what you mean by Textmate "eat"ing spaces.  But in both cases the spaces won't have specific tokens associated with them.

Reply:
To clarify what I meant by "eating"... 

An example of a simple grammar in extended Backus-Naur form could be this...

```
letter ::= "A" | "B" ;
number ::= "0" | "1" ;
identifier ::= letter , { letter | number };
program ::= { identifier };
```

Here, `ABBA10AB` would be a valid grammar, but not `ABBA10 AB` because whitespace must be explicitly accounted for (i.e., whitespace is not "eaten").

So, to get the above to work for the second form with spaces, you'd have to have the grammar something like this...

```
letter ::= "A" | "B" ;
number ::= "0" | "1" ;
whitespace ::= " " | "\t" | "\n" | "\r" ;
identifier ::= letter , { letter | number };
program ::= { identifier, whitespace };
```

.....

My point: Does TextMate "eat" whitespace is asking whether or not a grammar will lock up if it hits a whitespace character not explicitly accounted for.

Reply:
Done. ✅

Reply:
Not everything has to have scopes.  If it hits a character that doesn't match anything, it ignores it and moves onto the next character.

Reply:
Aha... Makes sense. Thanks for clarifying.